hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTestThreads.java

```diff
@@ -0,0 +1,11 @@
+package com.hazelcast.mapreduce;
+
+import com.hazelcast.test.MaxThreadsAware;
+import com.hazelcast.test.TestEnvironment;
+
+public class MapReduceTestThreads implements MaxThreadsAware {
+    @Override
+    public int maxThreads() {
+        return !TestEnvironment.isMockNetwork() ? 1 : Math.max(Runtime.getRuntime().availableProcessors() / 2, 1);
```

Comment:
Implementations of `MaxThreadsAware` should not deal with the `TestEnvironment.isMockNetwork()`. It should remain a responsibility of the ParallelRunner. 

When the MockNetwork is disabled then parallel runner should always use 1 thread.